src/NuGet.Core/NuGet.PackageManagement/PublicAPI.Unshipped.txt

```diff
@@ -1 +1,2 @@
 NuGet.PackageManagement.NuGetPackageManager.PreviewProjectsUninstallPackageAsync(System.Collections.Generic.IReadOnlyCollection<NuGet.ProjectManagement.NuGetProject> nuGetProjects, string packageId, NuGet.PackageManagement.UninstallationContext uninstallationContext, NuGet.ProjectManagement.INuGetProjectContext nuGetProjectContext, System.Threading.CancellationToken token) -> System.Threading.Tasks.Task<System.Collections.Generic.IEnumerable<NuGet.PackageManagement.NuGetProjectAction>>
+static NuGet.PackageManagement.NuGetPackageManager.GetGlobalPackageFolderRepositories(NuGet.Protocol.Core.Types.ISourceRepositoryProvider sourceRepositoryProvider, NuGet.Configuration.ISettings settings) -> System.Collections.Generic.IReadOnlyList<NuGet.Protocol.Core.Types.SourceRepository>
```

Comment:
Wearing my SDK owner hat, I don't think this is a good public API. 

Firstly, given the types and method name, it sounds like something that belongs in `NuGet.Protocol`, not `NuGet.PackageManagement`.

Secondly, looking at where it's used, it appears to be a way to avoid doing `var packageManager = new NuGetPackageManager(...); return packageManager.GlobalPackageFolderRepositories;`. Unless we have performance traces that show this is an actual performance issue, I don't think we should take on the technical debt of another public API for this.

Reply:
i'll be running with this PR now...taking a look at this feedback.

Reply:
ok, address this feedback. Please take a look at latest iteration.